### ISSUE TITLE
GraphQL Lacking Stale Content Delivery Time Implementation

### ISSUE THREAD

**Issue by AntonEvers** [fastly/fastly-magento2#559]
*As a GraphQL consumer using Fastly I want to receive stale content for a specific amount of time after cache invalidation so that I get fast responses more often and so that I do not have to wait for a slower origin request every time cache is invalidated, also in case of server failure.*

### AC
* `stale-while-revalidate` will be sent in GraphQL response header `cache-control` from Adobe Commerce 
* The value of `stale-while-revalidate` will be equal to the configuration field `full_page_cache/fastly/fastly_advanced_configuration/stale_ttl` titled **Stale Content Delivery Time**
* `stale-if-error` will be sent in GraphQL response header `cache-control` from Adobe Commerce 
* The value of `stale-if-error` will be equal to the configuration field `full_page_cache/fastly/fastly_advanced_configuration/stale_error_ttl` titled **Stale Content Delivery Time in Case of Backend Error**
* This functionality should work the same as `\Fastly\Cdn\Model\Layout\LayoutPlugin::afterGenerateElements` but then for GraphQL requests.

### Approved Schema
* none

### Technical direction
Recommendation to build this as a plugin to \Magento\Framework\App\Response\Http::setPublicHeaders, right after \Magento\GraphQlCache\Controller\Plugin\GraphQl::afterRenderResult is called.
See vendor/magento/module-graph-ql-cache/etc/graphql/di.xml as example:

```xml
<type name="Magento\Framework\App\FrontControllerInterface">
    <plugin name="graphql-dispatch-plugin" type="Magento\GraphQlCache\Controller\Plugin\GraphQl"/>
    <plugin name="front-controller-builtin-cache" type="Magento\PageCache\Model\App\FrontController\BuiltinPlugin"/>
    <plugin name="front-controller-varnish-cache" type="Magento\PageCache\Model\App\FrontController\VarnishPlugin"/>
</type>
```

Note that this plugin only applies to the GraphQL area.

**Comment by vvuksan** [comment]
Released in 1.2.186.

**Comment by AntonEvers** [comment]
@vvuksan please see my comments on the commit, this PR is only half the solution to the mentioned problem, as `stale-if-error` is still missing.

**Comment by dpotkoc** [comment]
@AntonEvers stale-if-error will be added.

**Comment by vvuksan** [comment]
This has been released in 1.2.187.
